// react-forminate/dist/index-DN8rpHta.js
var Xr = Object.defineProperty;
var Jr = (e, t, r) => t in e ? Xr(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r;
var Ve = (e, t, r) => Jr(e, typeof t != "symbol" ? t + "" : t, r);
import H, { useState as le, useRef as br, useEffect as Ge, useMemo as Zr, createContext as Qr, useContext as ft, useDebugValue as Gt, createElement as en, memo as tn, lazy as te, Suspense as rn } from "react";
var Ze = { exports: {} }, qe = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ut;
function nn() {
  if (Ut) return qe;
  Ut = 1;
  var e = Symbol.for("react.transitional.element"), t = Symbol.for("react.fragment");
  function r(n, s, i) {
    var a = null;
    if (i !== void 0 && (a = "" + i), s.key !== void 0 && (a = "" + s.key), "key" in s) {
      i = {};
      for (var c in s)
        c !== "key" && (i[c] = s[c]);
    } else i = s;
    return s = i.ref, {
      $$typeof: e,
      type: n,
      key: a,
      ref: s !== void 0 ? s : null,
      props: i
    };
  }
  return qe.Fragment = t, qe.jsx = r, qe.jsxs = r, qe;
}
var Ye = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Kt;
function sn() {
  return Kt || (Kt = 1, process.env.NODE_ENV !== "production" && function() {
    function e(o) {
      if (o == null) return null;
      if (typeof o == "function")
        return o.$$typeof === Ne ? null : o.displayName || o.name || null;
      if (typeof o == "string") return o;
      switch (o) {
        case S:
          return "Fragment";
        case x:
          return "Portal";
        case M:
          return "Profiler";
        case v:
          return "StrictMode";
        case V:
          return "Suspense";
        case L:
          return "SuspenseList";
      }
      if (typeof o == "object")
        switch (typeof o.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), o.$$typeof) {
          case j:
            return (o.displayName || "Context") + ".Provider";
          case I:
            return (o._context.displayName || "Context") + ".Consumer";
          case R:
            var m = o.render;
            return o = o.displayName, o || (o = m.displayName || m.name || "", o = o !== "" ? "ForwardRef(" + o + ")" : "ForwardRef"), o;
          case U:
            return m = o.displayName || null, m !== null ? m : e(o.type) || "Memo";
          case Z:
            m = o._payload, o = o._init;
            try {
              return e(o(m));
            } catch {
            }
        }
      return null;
    }
    function t(o) {
      return "" + o;
    }
    function r(o) {
      try {
        t(o);
        var m = !1;
      } catch {
        m = !0;
      }
      if (m) {
        m = console;
        var g = m.error, N = typeof Symbol == "function" && Symbol.toStringTag && o[Symbol.toStringTag] || o.constructor.name || "Object";
        return g.call(
          m,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          N
        ), t(o);
      }
    }
    function n() {
    }
    function s() {
      if (se === 0) {
        ge = console.log, de = console.info, ye = console.warn, oe = console.error, re = console.group, Fe = console.groupCollapsed, ve = console.groupEnd;
        var o = {
          configurable: !0,
          enumerable: !0,
          value: n,
          writable: !0
        };
        Object.defineProperties(console, {
          info: o,
          log: o,
          warn: o,
          error: o,
          group: o,
          groupCollapsed: o,
          groupEnd: o
        });
      }
      se++;
    }
    function i() {
      if (se--, se === 0) {
        var o = { configurable: !0, enumerable: !0, writable: !0 };
        Object.defineProperties(console, {
          log: K({}, o, { value: ge }),
          info: K({}, o, { value: de }),
          warn: K({}, o, { value: ye }),
          error: K({}, o, { value: oe }),
          group: K({}, o, { value: re }),
          groupCollapsed: K({}, o, { value: Fe }),
          groupEnd: K({}, o, { value: ve })
        });
      }
      0 > se && console.error(
        "disabledDepth fell below zero. This is a bug in React. Please file an issue."
      );
    }
    function a(o) {
      if (X === void 0)
        try {
          throw Error();
        } catch (g) {
          var m = g.stack.trim().match(/\n( *(at )?)/);
          X = m && m[1] || "", Bt = -1 < g.stack.indexOf(`
    at`) ? " (<anonymous>)" : -1 < g.stack.indexOf("@") ? "@unknown:0:0" : "";
        }
      return `
` + X + o + Bt;
    }
    function c(o, m) {
      if (!o || wt) return "";
      var g = St.get(o);
      if (g !== void 0) return g;
      wt = !0, g = Error.prepareStackTrace, Error.prepareStackTrace = void 0;
      var N = null;
      N = B.H, B.H = null, s();
      try {
        var q = {
          DetermineComponentFrameRoot: function() {
            try {
              if (m) {
                var fe = function() {
                  throw Error();
                };
                if (Object.defineProperty(fe.prototype, "props", {
                  set: function() {
                    throw Error();
                  }
                }), typeof Reflect == "object" && Reflect.construct) {
                  try {
                    Reflect.construct(fe, []);
                  } catch (ae) {
                    var Je = ae;
                  }
                  Reflect.construct(o, [], fe);
                } else {
                  try {
                    fe.call();
                  } catch (ae) {
                    Je = ae;
                  }
                  o.call(fe.prototype);
                }
              } else {
                try {
                  throw Error();
                } catch (ae) {
                  Je = ae;
                }
                (fe = o()) && typeof fe.catch == "function" && fe.catch(function() {
                });
              }
            } catch (ae) {
              if (ae && Je && typeof ae.stack == "string")
                return [ae.stack, Je.stack];
            }
            return [null, null];
          }
        };
        q.DetermineComponentFrameRoot.displayName = "DetermineComponentFrameRoot";
        var D = Object.getOwnPropertyDescriptor(
          q.DetermineComponentFrameRoot,
          "name"
        );
        D && D.configurable && Object.defineProperty(
          q.DetermineComponentFrameRoot,
          "name",
          { value: "DetermineComponentFrameRoot" }
        );
        var O = q.DetermineComponentFrameRoot(), ie = O[0], Re = O[1];
        if (ie && Re) {
          var Y = ie.split(`
`), be = Re.split(`
`);
          for (O = D = 0; D < Y.length && !Y[D].includes(
            "DetermineComponentFrameRoot"
          ); )
            D++;
          for (; O < be.length && !be[O].includes(
            "DetermineComponentFrameRoot"
          ); )
            O++;
          if (D === Y.length || O === be.length)
            for (D = Y.length - 1, O = be.length - 1; 1 <= D && 0 <= O && Y[D] !== be[O]; )
              O--;
          for (; 1 <= D && 0 <= O; D--, O--)
            if (Y[D] !== be[O]) {
              if (D !== 1 || O !== 1)
                do
                  if (D--, O--, 0 > O || Y[D] !== be[O]) {
                    var Be = `
` + Y[D].replace(
                      " at new ",
                      " at "
                    );
                    return o.displayName && Be.includes("<anonymous>") && (Be = Be.replace("<anonymous>", o.displayName)), typeof o == "function" && St.set(o, Be), Be;
                  }
                while (1 <= D && 0 <= O);
              break;
            }
        }
      } finally {
        wt = !1, B.H = N, i(), Error.prepareStackTrace = g;
      }
      return Y = (Y = o ? o.displayName || o.name : "") ? a(Y) : "", typeof o == "function" && St.set(o, Y), Y;
    }
    function l(o) {
      if (o == null) return "";
      if (typeof o == "function") {
        var m = o.prototype;
        return c(
          o,
          !(!m || !m.isReactComponent)
        );
      }
      if (typeof o == "string") return a(o);
      switch (o) {
        case V:
          return a("Suspense");
        case L:
          return a("SuspenseList");
      }
      if (typeof o == "object")
        switch (o.$$typeof) {
          case R:
            return o = c(o.render, !1), o;
          case U:
            return l(o.type);
          case Z:
            m = o._payload, o = o._init;
            try {
              return l(o(m));
            } catch {
            }
        }
      return "";
    }
    function d() {
      var o = B.A;
      return o === null ? null : o.getOwner();
    }
    function u(o) {
      if (ue.call(o, "key")) {
        var m = Object.getOwnPropertyDescriptor(o, "key").get;
        if (m && m.isReactWarning) return !1;
      }
      return o.key !== void 0;
    }
    function b(o, m) {
      function g() {
        qt || (qt = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          m
        ));
      }
      g.isReactWarning = !0, Object.defineProperty(o, "key", {
        get: g,
        configurable: !0
      });
    }
    function y() {
      var o = e(this.type);
      return Yt[o] || (Yt[o] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), o = this.props.ref, o !== void 0 ? o : null;
    }
    function f(o, m, g, N, q, D) {
      return g = D.ref, o = {
        $$typeof: p,
        type: o,
        key: m,
        props: D,
        _owner: q
      }, (g !== void 0 ? g : null) !== null ? Object.defineProperty(o, "ref", {
        enumerable: !1,
        get: y
      }) : Object.defineProperty(o, "ref", { enumerable: !1, value: null }), o._store = {}, Object.defineProperty(o._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(o, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.freeze && (Object.freeze(o.props), Object.freeze(o)), o;
    }
    function w(o, m, g, N, q, D) {
      if (typeof o == "string" || typeof o == "function" || o === S || o === M || o === v || o === V || o === L || o === Te || typeof o == "object" && o !== null && (o.$$typeof === Z || o.$$typeof === U || o.$$typeof === j || o.$$typeof === I || o.$$typeof === R || o.$$typeof === W || o.getModuleId !== void 0)) {
        var O = m.children;
        if (O !== void 0)
          if (N)
            if (me(O)) {
              for (N = 0; N < O.length; N++)
                k(O[N], o);
              Object.freeze && Object.freeze(O);
            } else
              console.error(
                "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
              );
          else k(O, o);
      } else
        O = "", (o === void 0 || typeof o == "object" && o !== null && Object.keys(o).length === 0) && (O += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports."), o === null ? N = "null" : me(o) ? N = "array" : o !== void 0 && o.$$typeof === p ? (N = "<" + (e(o.type) || "Unknown") + " />", O = " Did you accidentally export a JSX literal instead of a component?") : N = typeof o, console.error(
          "React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s",
          N,
          O
        );
      if (ue.call(m, "key")) {
        O = e(o);
        var ie = Object.keys(m).filter(function(Y) {
          return Y !== "key";
        });
        N = 0 < ie.length ? "{key: someKey, " + ie.join(": ..., ") + ": ...}" : "{key: someKey}", Wt[O + N] || (ie = 0 < ie.length ? "{" + ie.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          N,
          O,
          ie,
          O
        ), Wt[O + N] = !0);
      }
      if (O = null, g !== void 0 && (r(g), O = "" + g), u(m) && (r(m.key), O = "" + m.key), "key" in m) {
        g = {};
        for (var Re in m)
          Re !== "key" && (g[Re] = m[Re]);
      } else g = m;
      return O && b(
        g,
        typeof o == "function" ? o.displayName || o.name || "Unknown" : o
      ), f(o, O, D, q, d(), g);
    }
    function k(o, m) {
      if (typeof o == "object" && o && o.$$typeof !== Kr) {
        if (me(o))
          for (var g = 0; g < o.length; g++) {
            var N = o[g];
            C(N) && A(N, m);
          }
        else if (C(o))
          o._store && (o._store.validated = 1);
        else if (o === null || typeof o != "object" ? g = null : (g = he && o[he] || o["@@iterator"], g = typeof g == "function" ? g : null), typeof g == "function" && g !== o.entries && (g = g.call(o), g !== o))
          for (; !(o = g.next()).done; )
            C(o.value) && A(o.value, m);
      }
    }
    function C(o) {
      return typeof o == "object" && o !== null && o.$$typeof === p;
    }
    function A(o, m) {
      if (o._store && !o._store.validated && o.key == null && (o._store.validated = 1, m = E(m), !Ht[m])) {
        Ht[m] = !0;
        var g = "";
        o && o._owner != null && o._owner !== d() && (g = null, typeof o._owner.tag == "number" ? g = e(o._owner.type) : typeof o._owner.name == "string" && (g = o._owner.name), g = " It was passed a child from " + g + ".");
        var N = B.getCurrentStack;
        B.getCurrentStack = function() {
          var q = l(o.type);
          return N && (q += N() || ""), q;
        }, console.error(
          'Each child in a list should have a unique "key" prop.%s%s See https://react.dev/link/warning-keys for more information.',
          m,
          g
        ), B.getCurrentStack = N;
      }
    }
    function E(o) {
      var m = "", g = d();
      return g && (g = e(g.type)) && (m = `

Check the render method of \`` + g + "`."), m || (o = e(o)) && (m = `

Check the top-level render call using <` + o + ">."), m;
    }
    var h = H, p = Symbol.for("react.transitional.element"), x = Symbol.for("react.portal"), S = Symbol.for("react.fragment"), v = Symbol.for("react.strict_mode"), M = Symbol.for("react.profiler"), I = Symbol.for("react.consumer"), j = Symbol.for("react.context"), R = Symbol.for("react.forward_ref"), V = Symbol.for("react.suspense"), L = Symbol.for("react.suspense_list"), U = Symbol.for("react.memo"), Z = Symbol.for("react.lazy"), Te = Symbol.for("react.offscreen"), he = Symbol.iterator, Ne = Symbol.for("react.client.reference"), B = h.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, ue = Object.prototype.hasOwnProperty, K = Object.assign, W = Symbol.for("react.client.reference"), me = Array.isArray, se = 0, ge, de, ye, oe, re, Fe, ve;
    n.__reactDisabledLog = !0;
    var X, Bt, wt = !1, St = new (typeof WeakMap == "function" ? WeakMap : Map)(), Kr = Symbol.for("react.client.reference"), qt, Yt = {}, Wt = {}, Ht = {};
    Ye.Fragment = S, Ye.jsx = function(o, m, g, N, q) {
      return w(o, m, g, !1, N, q);
    }, Ye.jsxs = function(o, m, g, N, q) {
      return w(o, m, g, !0, N, q);
    };
  }()), Ye;
}
var Xt;
function on() {
  return Xt || (Xt = 1, process.env.NODE_ENV === "production" ? Ze.exports = nn() : Ze.exports = sn()), Ze.exports;
}
var P = on();
class an {
  constructor() {
    Ve(this, "subscribers", /* @__PURE__ */ new Map());
  }
  subscribe(t, r) {
    return this.subscribers.has(t) || this.subscribers.set(t, /* @__PURE__ */ new Set()), this.subscribers.get(t).add(r), () => this.subscribers.get(t)?.delete(r);
  }
  notify(t) {
    const r = this.subscribers.get(t);
    r && r.forEach((n) => n());
  }
  unsubscribeAll(t) {
    this.subscribers.delete(t);
  }
  clear() {
    this.subscribers.clear();
  }
}
const cn = [
  // React-specific
  "className",
  "htmlFor",
  "defaultValue",
  "defaultChecked",
  "onChange",
  "onClick",
  "onBlur",
  "onFocus",
  "onKeyDown",
  "onKeyUp",
  "onMouseEnter",
  "onMouseLeave",
  "onInput",
  "onSubmit",
  "onInvalid",
  "children",
  "key",
  "ref",
  "style",
  "dangerouslySetInnerHTML",
  "suppressContentEditableWarning",
  "suppressHydrationWarning",
  // Accessibility / ARIA
  "role",
  "tabIndex",
  "aria-hidden",
  "aria-label",
  "aria-labelledby",
  "aria-describedby",
  "aria-live",
  "aria-expanded",
  "aria-checked",
  "aria-disabled",
  "aria-required",
  // Data attributes
  "data-testid",
  "data-*",
  // Form/input related
  "name",
  "type",
  "value",
  "placeholder",
  "checked",
  "disabled",
  "readOnly",
  "required",
  "min",
  "max",
  "minLength",
  "maxLength",
  "pattern",
  "step",
  "multiple",
  "accept",
  "autoFocus",
  "autoComplete",
  "autoCorrect",
  "spellCheck",
  "form",
  "formAction",
  "formMethod",
  "formNoValidate",
  "formTarget",
  "inputMode",
  "list",
  // Common HTML attributes
  "id",
  "title",
  "hidden",
  "dir",
  "lang",
  "contentEditable",
  "draggable",
  "accessKey",
  // Media
  "src",
  "alt",
  "width",
  "height",
  "poster",
  "preload",
  "controls",
  "muted",
  "loop",
  "playsInline",
  // Anchor/link
  "href",
  "target",
  "download",
  "rel",
  // Misc
  "dangerouslySetInnerHTML"
  // Add more if needed
], ln = {
  input: "input",
  textarea: "textarea",
  select: "select",
  button: "button",
  form: "form",
  div: "div",
  span: "span",
  label: "label",
  a: "a",
  img: "img",
  radio: "input",
  checkbox: "input",
  file: "input"
  // Add more if needed
}, un = (e, t, r = {}, n) => Object.entries(e).reduce((s, [i, a]) => {
  const c = yn(a, {
    fieldId: t,
    values: r,
    fieldSchema: e,
    formSchema: n
  });
  return {
    ...s,
    [i]: c
  };
}, {}), dn = (e) => {
  const t = [], n = e.toString().match(
    /(values|formValues|values\?)\.([a-zA-Z0-9_]+)/g
  );
  return n && n.forEach((s) => {
    const i = s.split(".")[1];
    t.push(i);
  }), t;
}, wr = (e) => typeof e == "function" ? {
  fn: e,
  dependsOn: [...dn(e)]
} : e, fn = (e, t = {}) => {
  function r(i) {
    const a = [];
    for (const c in i)
      typeof i[c] == "object" && i[c] !== null && Array.isArray(i[c]?.dependsOn) && a.push(...i[c].dependsOn);
    return a;
  }
  function n(i) {
    for (const a of i) {
      if (!a.fieldId) continue;
      const c = r(a);
      for (const l of c)
        t[l] || (t[l] = /* @__PURE__ */ new Set()), t[l].add(a.fieldId);
      Array.isArray(a.fields) && n(a.fields);
    }
  }
  n(e);
  const s = {};
  for (const i in t)
    s[i] = Array.from(t[i]);
  return s;
}, pn = (e) => {
  const t = {}, r = (n) => {
    !n || n.length === 0 || n.forEach((s) => {
      s.dynamicOptions?.dependsOn && (t[s.fieldId] = s.dynamicOptions.dependsOn), s.fields && s.fields.length > 0 && r(s.fields);
    });
  };
  return r(e), t;
}, hn = {
  checkbox: [],
  number: 0,
  tel: "",
  text: "",
  email: "",
  url: "",
  password: "",
  search: "",
  date: "",
  radio: "",
  select: "",
  textarea: "",
  gridview: [],
  container: {},
  spacer: "",
  group: []
  // Add other field types as needed
};
function mn(e) {
  return e.type === "select";
}
function gn(e, t, r = []) {
  const n = document.createElement(ln[e] || "div"), s = /* @__PURE__ */ new Set();
  for (const a in n)
    s.add(a);
  cn?.forEach((a) => s.add(a));
  const i = (a) => !r.includes(a) && (s.has(a) || a.startsWith("data-") || a.startsWith("aria-"));
  return Object.fromEntries(
    Object.entries(t).filter(([a]) => i(a))
  );
}
const yn = (e, t, r) => {
  if (e && typeof e == "object" && "fn" in e && typeof e.fn == "function")
    try {
      return e.fn({
        fieldId: t?.fieldId,
        values: t?.values,
        fieldSchema: t.fieldSchema,
        //getFieldSchema(functionParams?.fieldId),
        formSchema: t?.formSchema
      });
    } catch (n) {
      return console.error("Error in field prop function:", n), r;
    }
  if (typeof e == "function")
    try {
      return e({ ...t });
    } catch (n) {
      return console.error("Error in field prop function:", n), r;
    }
  return e !== void 0 ? e : r;
}, pt = (e, t = [], r = {}, n) => {
  for (const s of t) {
    if (s.fieldId === e)
      return s;
    if (s.fields && s.fields.length > 0) {
      const i = pt(e, s.fields, r);
      if (i)
        return i;
    }
  }
  return null;
}, jt = (e) => {
  if (typeof e == "number") return !0;
  if (typeof e == "string") {
    const t = e.trim();
    return !isNaN(t) && !isNaN(parseFloat(t));
  }
  return !1;
}, xe = class xe {
  constructor() {
    Ve(this, "cache", /* @__PURE__ */ new Map());
  }
  static getInstance() {
    return xe.instance || (xe.instance = new xe()), xe.instance;
  }
  process(t, r, n) {
    const s = this.getCacheKey(t, r);
    if (this.cache.has(s))
      return this.cache.get(s);
    const i = this.processField(t, r, n);
    return this.cache.set(s, i), i;
  }
  processAllFields(t, r, n) {
    return t.map((s) => this.process(s, r, n));
  }
  processField(t, r, n) {
    const s = {
      fieldId: t.fieldId,
      values: r,
      fieldSchema: t,
      formSchema: n
    }, i = zn.processFieldProps(t, s);
    return t.fields && t.fields.length > 0 && (i.fields = t.fields.map(
      (a) => this.processField(a, r, n)
    )), i;
  }
  getCacheKey(t, r) {
    const s = this.getFieldDependencies(t).map((i) => `${i}:${JSON.stringify(r[i])}`).join("|");
    return `${t.fieldId}|${s}|${JSON.stringify(t)}`;
  }
  getFieldDependencies(t) {
    const r = /* @__PURE__ */ new Set();
    return [
      "label",
      "required",
      "disabled",
      "visibility",
      "options",
      "requiredMessage"
    ].forEach((s) => {
      const i = t[s], a = wr(
        i
      );
      a && typeof a == "object" && "dependsOn" in a && Array.isArray(a.dependsOn) && a.dependsOn.forEach(
        (c) => r.add(c)
      );
    }), typeof t.visibility == "object" && "dependsOn" in t.visibility && t.visibility.dependsOn && (Array.isArray(t.visibility.dependsOn) && t.visibility.dependsOn.length ? t.visibility.dependsOn.forEach((s) => {
      r.add(s);
    }) : r.add(t.fieldId)), Array.from(r);
  }
  clearCache() {
    this.cache.clear();
  }
  clearCacheForField(t) {
    Array.from(this.cache.keys()).filter((r) => r.startsWith(`${t}|`)).forEach((r) => this.cache.delete(r));
  }
};
Ve(xe, "instance");
let Ue = xe;
const vn = (e, t, r, n, s, i, a = !1) => {
  const c = Ue.getInstance(), l = pt(
    e,
    r.fields,
    n
  );
  if (!l || r.options?.validateFieldsOnBlur !== !1 && !a && i && !i[e])
    return;
  const u = Dt(l, n), b = Sr(l, n, r);
  if (!u || b) {
    s((h) => {
      const p = { ...h };
      return delete p[e], p;
    });
    return;
  }
  if (!a && i && !i[e])
    return;
  const y = c.process(l, n, r), w = ["checkbox", "radio"].includes(l.type) && Array.isArray(t) && t.length === 0, k = t === "" || t === null || t === void 0 || w;
  if (k && !y.required) {
    s((h) => {
      const p = { ...h };
      return delete p[e], p;
    });
    return;
  }
  const C = [];
  if (y.required && C.push({
    type: "required",
    message: y.requiredMessage || "This field is required."
  }), !k && Array.isArray(y.validation) && C.push(...y.validation), C.length === 0) {
    s((h) => {
      const p = { ...h };
      return delete p[e], p;
    });
    return;
  }
  const { isValid: A, message: E } = xr.validate(
    t,
    C
  );
  s((h) => {
    if (!A)
      return { ...h, [e]: E || "Validation failed" };
    const p = { ...h };
    return delete p[e], p;
  });
}, bn = (e, t, r, n, s = !1) => {
  const i = Ue.getInstance();
  let a = !0;
  const c = {}, l = (d) => {
    !d || d.length === 0 || d.forEach((u) => {
      if (u.fields && u.fields.length > 0) {
        if (!u.fieldId)
          throw new Error("`fieldId` is not provided for the field");
        l(u.fields);
        return;
      }
      const b = Dt(u, t), y = Sr(u, t, e);
      if (!b || y || !s && n && !n[u.fieldId])
        return;
      const f = i.process(u, t, e), w = t[u.fieldId], C = ["checkbox", "radio"].includes(u.type) && Array.isArray(w) && w.length === 0, A = w === "" || w === null || w === void 0 || C;
      if (A && !f.required)
        return;
      const E = [];
      if (f.required && E.push({
        type: "required",
        message: f.requiredMessage || "This field is required."
      }), !A && Array.isArray(f.validation) && E.push(...f.validation), E.length === 0)
        return;
      const { isValid: h, message: p } = xr.validate(
        w,
        E
      );
      h || (c[u.fieldId] = p || "Validation failed", a = !1);
    });
  };
  return r({}), l(e.fields), r(c), a;
}, Dt = (e, t) => {
  const r = e.visibility;
  if (typeof r > "u" || r === !0)
    return !0;
  if (typeof r == "boolean") return r;
  if (typeof r == "object" && "dependsOn" in r && "condition" in r && "value" in r) {
    const { dependsOn: n, condition: s, value: i } = r, a = t[n];
    switch (s) {
      case "equals":
        return a === i;
      case "not_equals":
        return a !== i;
      default:
        return !0;
    }
  }
  return !0;
}, Sr = (e, t, r) => {
  if (typeof e.disabled == "boolean") return e.disabled;
  if (typeof e.disabled > "u") return !1;
  if (typeof e.disabled == "object" && "fn" in e.disabled && typeof e.disabled.fn == "function")
    return e.disabled.fn({
      fieldId: e?.fieldId,
      values: t,
      fieldSchema: e,
      formSchema: r
    });
  if (typeof e.disabled == "function") {
    const n = un(
      e,
      e.fieldId,
      t,
      r
    );
    return n && n.disabled || !1;
  }
  return !!e.disabled;
};
class wn {
  validate(t, r) {
    if (typeof t != "string")
      return { isValid: !1, message: "Value must be a string." };
    if (r.pattern === void 0)
      return { isValid: !0 };
    const s = new RegExp(r.pattern || "").test(t);
    return {
      isValid: s,
      message: s ? void 0 : r.message || "Pattern validation failed."
    };
  }
}
class Sn {
  validate(t, r) {
    if (typeof t != "string")
      return { isValid: !1, message: "Value must be a string." };
    if (r.minLength === void 0)
      return { isValid: !0 };
    const n = t.length >= r.minLength;
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Minimum length is ${r.minLength}.`
    };
  }
}
class xn {
  validate(t, r) {
    if (typeof t != "string")
      return { isValid: !1, message: "Value must be a string." };
    if (r.maxLength === void 0)
      return { isValid: !0 };
    const n = t.length <= r.maxLength;
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Maximum length is ${r.maxLength}.`
    };
  }
}
class En {
  validate(t, r) {
    if (jt(t) === !1)
      return { isValid: !1, message: "Value must be a number." };
    if (r.min === void 0)
      return { isValid: !0 };
    const s = (typeof t == "string" ? parseFloat(t) : t) >= r.min;
    return {
      isValid: s,
      message: s ? void 0 : r.message || `Minimum value is ${r.min}`
    };
  }
}
class kn {
  validate(t, r) {
    if (jt(t) === !1)
      return { isValid: !1, message: "Value must be a number." };
    if (r.max === void 0)
      return { isValid: !0 };
    const s = (typeof t == "string" ? parseFloat(t) : t) <= r.max;
    return {
      isValid: s,
      message: s ? void 0 : r.message || `Maximum value is ${r.max}`
    };
  }
}
class Cn {
  validate(t, r) {
    if (jt(t) === !1)
      return { isValid: !1, message: "Value must be a number." };
    const n = typeof t == "string" ? parseFloat(t) : t;
    let s = !0;
    return r.min !== void 0 && n < r.min && (s = !1), r.max !== void 0 && n > r.max && (s = !1), {
      isValid: s,
      message: s ? void 0 : r.message
    };
  }
}
class An {
  validate(t, r) {
    if (typeof r.custom != "function")
      return { isValid: !0 };
    const n = r.custom(t);
    return {
      isValid: n,
      message: n ? void 0 : r.message || "Custom validation failed."
    };
  }
}
class On {
  validate(t, r) {
    const n = t != null && t !== "" && t.length !== 0;
    return {
      isValid: n,
      message: n ? void 0 : r.message || "This field is required."
    };
  }
}
class Tn {
  validate(t, r) {
    if (!(t instanceof Date))
      return { isValid: !1, message: "Value must be a date." };
    if (r.minDate === void 0)
      return { isValid: !0 };
    const n = t >= new Date(r.minDate);
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Date must be after ${r.minDate}.`
    };
  }
}
class Nn {
  validate(t, r) {
    if (!(t instanceof Date))
      return { isValid: !1, message: "Value must be a date." };
    if (r.maxDate === void 0)
      return { isValid: !0 };
    const n = t <= new Date(r.maxDate);
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Date must be before ${r.maxDate}.`
    };
  }
}
class Fn {
  validate(t, r) {
    if (!(t instanceof Date))
      return { isValid: !1, message: "Value must be a date." };
    let n = !0;
    if (r.minDate !== void 0) {
      const s = new Date(r.minDate);
      t < s && (n = !1);
    }
    if (r.maxDate !== void 0) {
      const s = new Date(r.maxDate);
      t > s && (n = !1);
    }
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Date must be between ${r.minDate} and ${r.maxDate}.`
    };
  }
}
class Rn {
  validate(t, r) {
    if (!Array.isArray(t))
      return { isValid: !1, message: "Value must be an array." };
    if (r.minItems === void 0)
      return { isValid: !0 };
    const n = t.length >= r.minItems;
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Minimum items required is ${r.minItems}.`
    };
  }
}
class Vn {
  validate(t, r) {
    if (!Array.isArray(t))
      return { isValid: !1, message: "Value must be an array." };
    if (r.maxItems === void 0)
      return { isValid: !0 };
    const n = t.length <= r.maxItems;
    return {
      isValid: n,
      message: n ? void 0 : r.message || `Maximum items allowed is ${r.maxItems}.`
    };
  }
}
const Ee = class Ee {
  constructor() {
    Ve(this, "strategies", {});
    this.strategies = {}, this.registerDefaultStrategies();
  }
  registerDefaultStrategies() {
    this.registerStrategy("pattern", new wn()), this.registerStrategy("minLength", new Sn()), this.registerStrategy("maxLength", new xn()), this.registerStrategy("min", new En()), this.registerStrategy("max", new kn()), this.registerStrategy("range", new Cn()), this.registerStrategy("custom", new An()), this.registerStrategy("required", new On()), this.registerStrategy("minDate", new Tn()), this.registerStrategy("maxDate", new Nn()), this.registerStrategy("dateRange", new Fn()), this.registerStrategy("minItems", new Rn()), this.registerStrategy("maxItems", new Vn());
  }
  determineRuleType(t) {
    return t.type ? t.type : t.min !== void 0 && t.max !== void 0 ? "range" : t.minDate !== void 0 && t.maxDate !== void 0 ? "dateRange" : t.pattern ? "pattern" : t.minLength !== void 0 ? "minLength" : t.maxLength !== void 0 ? "maxLength" : t.min !== void 0 ? "min" : t.max !== void 0 ? "max" : t.custom ? "custom" : t.minDate ? "minDate" : t.maxDate ? "maxDate" : t.minItems !== void 0 ? "minItems" : t.maxItems !== void 0 ? "maxItems" : "unknown";
  }
  registerStrategy(t, r) {
    this.strategies[t] = r;
  }
  validate(t, r) {
    if (!r || r.length === 0)
      return { isValid: !0 };
    for (const n of r) {
      const s = this.determineRuleType(n), i = this.strategies[s];
      if (i) {
        const a = i.validate(t, n);
        if (!a.isValid)
          return a;
      }
    }
    return { isValid: !0 };
  }
  static getInstance() {
    return Ee.instance || (Ee.instance = new Ee()), Ee.instance;
  }
};
Ve(Ee, "instance");
let At = Ee;
const xr = At.getInstance(), _n = (e) => {
  const [t, r] = le(
    {}
  ), n = br({});
  return { dynamicOptions: t, fetchDynamicOptions: async (i, a = {}, c) => {
    const d = pt(i, e.fields)?.dynamicOptions;
    if (!d || !d.endpoint) return;
    n.current[i]?.abort();
    const u = new AbortController();
    n.current[i] = u;
    let b = d.endpoint;
    b = b.replace(
      /\{\{(.*?)\}\}/g,
      (x, S) => a[S] ?? ""
    );
    const y = new URLSearchParams();
    if (d.params)
      for (const [x, S] of Object.entries(d.params)) {
        const v = a[S];
        v !== void 0 && y.append(x, v);
      }
    const f = d.pagination, w = c?.page ?? 1, k = c?.limit ?? f?.limit ?? 10, C = f?.startPage ?? 1;
    if ((f?.pageMode ?? "page") === "skip") {
      const x = f?.skipKey || "skip", S = (w - C) * k;
      y.set(x, String(S));
    } else {
      const x = f?.pageKey || "page";
      y.set(x, String(w));
    }
    const E = f?.limitKey || "limit";
    y.set(E, String(k));
    const h = y.toString() ? `${b}${b.includes("?") ? "&" : "?"}${y.toString()}` : b, p = async () => {
      if (f?.maxPage && w > f.maxPage)
        throw new Error("Max page limit reached");
      return (await fetch(h, {
        method: d.method || "GET",
        headers: d.headers || {},
        signal: u.signal
      })).json();
    };
    try {
      let x = await p().catch(() => (console.warn(`[Retry] Failed fetching ${i}, retrying...`), p()));
      if (d.resultPath)
        for (const v of d.resultPath.split("."))
          x = x?.[v];
      const S = d.transformResponse ? d.transformResponse(x) : x;
      r((v) => ({
        ...v,
        [i]: S
      }));
    } catch (x) {
      x.name === "AbortError" ? console.log(`[Aborted] Fetch for ${i} was cancelled.`) : console.error(`Failed to fetch dynamicOptions for ${i}`, x);
    }
  } };
}, Pn = (e, t) => {
  const { values: r, setValue: n } = Ke();
  Ge(() => {
    r[e] === void 0 && t !== void 0 && n(e, t);
  }, [e, r, t, n]);
}, jn = (e) => {
  const { values: t, formSchema: r } = Ke(), n = Ue.getInstance();
  return Zr(() => n.process(e, t, r), [e, t, r]);
}, Dn = (e) => {
  const {
    values: t,
    errors: r,
    formSchema: n,
    dynamicOptions: s,
    observer: i,
    formOptions: a = {},
    setValue: c,
    getFieldSchema: l,
    shouldShowField: d,
    validateForm: u,
    validateField: b,
    touched: y,
    setTouched: f
  } = Ke(), [w, k] = le(!1), C = e?.fieldId, A = t[C] || hn[e.type];
  Pn(C, e._defaultValue);
  let E;
  E = jn(e), Ge(() => {
    const I = i.subscribe(C, () => {
      console.log(`[${C}] Observer triggered!`);
    });
    return () => I();
  }, [C, i]);
  const h = (I) => {
    k(!0), e.events?.onCustomFocus && e.events.onCustomFocus(
      I,
      C,
      t,
      e,
      n
    );
  }, p = (I) => {
    f(e.fieldId, !0), w && a.validateFieldsOnBlur !== !1 && b(e.fieldId, t[e.fieldId]), e.events?.onCustomBlur && e.events.onCustomBlur(
      I,
      C,
      t,
      e,
      n
    );
  }, x = $n({
    fieldId: e.fieldId,
    type: e.type,
    value: A,
    ...e.events,
    onCustomBlur: p,
    // Override the blur handler
    onCustomFocus: h,
    // Override the focus handler
    onCustomUpload: e.events?.onCustomUpload
  }), S = Ln(e.type, E), v = d(E), M = E.disabled || !1;
  return {
    fieldId: C,
    processedProps: E,
    fieldParams: S,
    fieldValue: A,
    values: t,
    errors: r,
    formSchema: n,
    dynamicOptions: s,
    eventHandlers: x,
    isVisible: v,
    isDisable: M,
    observer: i,
    isTouched: y[e.fieldId] || !1,
    hasBeenFocused: w,
    setValue: c,
    getFieldSchema: l,
    validateForm: u,
    validateField: b
  };
}, In = () => {
  const {
    values: e,
    errors: t,
    formSchema: r,
    formOptions: n,
    touched: s,
    blurred: i,
    setBlurred: a,
    setValue: c,
    getFieldSchema: l,
    setTouched: d,
    validateField: u
  } = Ke();
  return {
    formOptions: n,
    touched: s,
    blurred: i,
    setValue: c,
    handleCustomEvent: (y, f, w, k) => {
      if (y) {
        const C = k ?? e[w], A = l(w);
        y(
          f,
          w,
          { ...e, [w]: C },
          A,
          r,
          t
        );
      }
    },
    setTouched: d,
    setBlurred: a,
    validateField: u
  };
}, $n = ({
  fieldId: e,
  value: t,
  type: r,
  onCustomChange: n,
  onCustomClick: s,
  onCustomBlur: i,
  onCustomFocus: a,
  onCustomKeyDown: c,
  onCustomKeyUp: l,
  onCustomMouseDown: d,
  onCustomMouseEnter: u,
  onCustomMouseLeave: b,
  onCustomContextMenu: y,
  onCustomUpload: f,
  onCustomRemove: w
}) => {
  const {
    formOptions: k,
    blurred: C,
    setBlurred: A,
    setValue: E,
    handleCustomEvent: h,
    setTouched: p,
    validateField: x
  } = In(), S = (V) => {
    p(e, !0), h(a, V, e);
  }, v = (V) => {
    A(e, !0), i && h(i, V, e), setTimeout(() => {
      k?.validateFieldsOnBlur !== !1 && x(e, t);
    }, 500);
  }, M = (V) => {
    const L = r === "number" ? +V.target.value : V.target.value;
    let U = L;
    if (r === "checkbox") {
      const Z = V.target.checked, Te = t || [];
      U = Z ? [...Te, L] : Te.filter((Ne) => Ne !== L);
    }
    E(e, U), (k?.validateFieldsOnBlur === !1 || C[e]) && x(e, L), h(n, V, e, U);
  }, I = (V) => {
    const L = Array.from((V instanceof FileList, V));
    f && f(L, e), E(e, L);
  }, j = (V) => {
    w && w(V, e);
  }, R = (V) => (L) => h(V, L, e, t);
  return {
    htmlHandlers: {
      onChange: M,
      onClick: R(s),
      onBlur: v,
      onFocus: S,
      onKeyDown: R(c),
      onKeyUp: R(l),
      onMouseDown: R(d),
      onMouseEnter: R(u),
      onMouseLeave: R(b),
      onContextMenu: R(y)
    },
    customHandlers: {
      onUpload: f ? I : void 0,
      onRemove: w ? j : void 0
    }
  };
}, Mn = ["required"], Q = (e = "input", t = "text") => ({
  "data-testid": `${e}-${t}-field`,
  excludedProps: [...Mn],
  htmlTagName: e
}), Jt = {
  text: { ...Q("input", "text") },
  email: { ...Q("input", "email") },
  password: { ...Q("input", "password") },
  number: { ...Q("input", "number") },
  tel: { ...Q("input", "tel") },
  url: { ...Q("input", "url") },
  search: { ...Q("input", "search") },
  radio: { ...Q("input", "radio") },
  checkbox: { ...Q("input", "checkbox") },
  select: { ...Q("select", "select") },
  file: { ...Q("input", "file") }
}, Ln = (e, t) => {
  const r = Jt[e] || Jt.text;
  return r.htmlTagName ? {
    ...gn(
      r.htmlTagName,
      t,
      r.excludedProps
    ),
    "data-testid": r["data-testid"],
    id: t.fieldId || t.id || ""
  } : {};
};
class zn {
  static process(t, r) {
    try {
      const n = wr(t) || t;
      return this.isComputedValue(n) ? n.fn(r) : n;
    } catch {
      return this.isComputedValue(t) ? t.fn(r) : t;
    }
  }
  static isComputedValue(t) {
    return t && typeof t == "object" && "fn" in t;
  }
  static processFieldProps(t, r) {
    const n = {};
    for (const [s, i] of Object.entries(t))
      this.isProcessableProperty(s) ? n[s] = this.process(i, r) : n[s] = i;
    return n;
  }
  static isProcessableProperty(t) {
    return [
      "label",
      "required",
      "disabled",
      "visibility",
      "className",
      "options",
      "placeholder",
      "validation",
      "requiredMessage"
    ].includes(t);
  }
}
const ct = Qr(
  void 0
), Ke = () => {
  const e = ft(ct);
  if (!e)
    throw new Error("useForm must be used within a FormProvider");
  return e;
}, Bn = ({
  children: e,
  formSchema: t
}) => {
  const [r, n] = le({}), [s, i] = le({}), [a] = le(t), [c, l] = le({}), [d, u] = le({}), b = pn(t.fields), { dynamicOptions: y, fetchDynamicOptions: f } = _n(t), w = ft(ct), k = new an(), C = fn(a.fields), { options: A = {} } = t;
  if (w)
    return /* @__PURE__ */ P.jsx(P.Fragment, { children: e });
  const E = (j, R) => {
    l((V) => ({ ...V, [j]: R }));
  }, h = (j, R) => {
    u((V) => ({ ...V, [j]: R }));
  }, p = (j, R) => {
    const V = { ...r, [j]: R };
    n(V);
    const L = C[j];
    L?.length && L.forEach((U) => {
      k.notify(U);
    }), Object.entries(b).forEach(([U, Z]) => {
      (Z === j || Array.isArray(Z) && Z.includes(j)) && f(U, V);
    });
  }, x = (j, R) => {
    vn(
      j,
      R,
      t,
      r,
      i,
      c,
      !1
    );
  }, S = () => bn(t, r, i, c, !0), v = (j) => pt(
    j,
    t.fields,
    r
  ), M = (j) => Dt(j, r), I = (j) => {
    for (const R of j)
      mn(R) && R.dynamicOptions?.fetchOnInit && f(R.fieldId, r), R.fields && R.fields.length > 0 && I(R.fields);
  };
  return Ge(() => {
    I(t.fields);
  }, []), /* @__PURE__ */ P.jsx(
    ct.Provider,
    {
      value: {
        values: r,
        errors: s,
        dynamicOptions: y,
        formSchema: a,
        observer: k,
        formOptions: A,
        touched: c,
        blurred: d,
        setTouched: E,
        setBlurred: h,
        setValue: p,
        validateField: x,
        validateForm: S,
        shouldShowField: M,
        fetchDynamicOptions: f,
        getFieldSchema: v
      },
      children: e
    }
  );
}, qn = H.createContext({}), Er = !0;
function Yn({ baseColor: e, highlightColor: t, width: r, height: n, borderRadius: s, circle: i, direction: a, duration: c, enableAnimation: l = Er, customHighlightBackground: d }) {
  const u = {};
  return a === "rtl" && (u["--animation-direction"] = "reverse"), typeof c == "number" && (u["--animation-duration"] = `${c}s`), l || (u["--pseudo-element-display"] = "none"), (typeof r == "string" || typeof r == "number") && (u.width = r), (typeof n == "string" || typeof n == "number") && (u.height = n), (typeof s == "string" || typeof s == "number") && (u.borderRadius = s), i && (u.borderRadius = "50%"), typeof e < "u" && (u["--base-color"] = e), typeof t < "u" && (u["--highlight-color"] = t), typeof d == "string" && (u["--custom-highlight-background"] = d), u;
}
function Zt({ count: e = 1, wrapper: t, className: r, containerClassName: n, containerTestId: s, circle: i = !1, style: a, ...c }) {
  var l, d, u;
  const b = H.useContext(qn), y = { ...c };
  for (const [h, p] of Object.entries(c))
    typeof p > "u" && delete y[h];
  const f = {
    ...b,
    ...y,
    circle: i
  }, w = {
    ...a,
    ...Yn(f)
  };
  let k = "react-loading-skeleton";
  r && (k += ` ${r}`);
  const C = (l = f.inline) !== null && l !== void 0 ? l : !1, A = [], E = Math.ceil(e);
  for (let h = 0; h < E; h++) {
    let p = w;
    if (E > e && h === E - 1) {
      const S = (d = p.width) !== null && d !== void 0 ? d : "100%", v = e % 1, M = typeof S == "number" ? S * v : `calc(${S} * ${v})`;
      p = { ...p, width: M };
    }
    const x = H.createElement("span", { className: k, style: p, key: h }, "‌");
    C ? A.push(x) : A.push(H.createElement(
      H.Fragment,
      { key: h },
      x,
      H.createElement("br", null)
    ));
  }
  return H.createElement("span", { className: n, "data-testid": s, "aria-live": "polite", "aria-busy": (u = f.enableAnimation) !== null && u !== void 0 ? u : Er }, t ? A.map((h, p) => H.createElement(t, { key: p }, h)) : A);
}
var G = function() {
  return G = Object.assign || function(t) {
    for (var r, n = 1, s = arguments.length; n < s; n++) {
      r = arguments[n];
      for (var i in r) Object.prototype.hasOwnProperty.call(r, i) && (t[i] = r[i]);
    }
    return t;
  }, G.apply(this, arguments);
};
function Pe(e, t, r) {
  if (r || arguments.length === 2) for (var n = 0, s = t.length, i; n < s; n++)
    (i || !(n in t)) && (i || (i = Array.prototype.slice.call(t, 0, n)), i[n] = t[n]);
  return e.concat(i || Array.prototype.slice.call(t));
}
function Wn(e) {
  var t = /* @__PURE__ */ Object.create(null);
  return function(r) {
    return t[r] === void 0 && (t[r] = e(r)), t[r];
  };
}
var Hn = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|disableRemotePlayback|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, Gn = /* @__PURE__ */ Wn(
  function(e) {
    return Hn.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
  /* Z+1 */
), _ = "-ms-", He = "-moz-", F = "-webkit-", kr = "comm", ht = "rule", It = "decl", Un = "@import", Cr = "@keyframes", Kn = "@layer", Ar = Math.abs, $t = String.fromCharCode, Ot = Object.assign;
function Xn(e, t) {
  return z(e, 0) ^ 45 ? (((t << 2 ^ z(e, 0)) << 2 ^ z(e, 1)) << 2 ^ z(e, 2)) << 2 ^ z(e, 3) : 0;
}
function Or(e) {
  return e.trim();
}
function ce(e, t) {
  return (e = t.exec(e)) ? e[0] : e;
}
function T(e, t, r) {
  return e.replace(t, r);
}
function nt(e, t, r) {
  return e.indexOf(t, r);
}
function z(e, t) {
  return e.charCodeAt(t) | 0;
}
function je(e, t, r) {
  return e.slice(t, r);
}
function ne(e) {
  return e.length;
}
function Tr(e) {
  return e.length;
}
function We(e, t) {
  return t.push(e), e;
}
function Jn(e, t) {
  return e.map(t).join("");
}
function Qt(e, t) {
  return e.filter(function(r) {
    return !ce(r, t);
  });
}
var mt = 1, De = 1, Nr = 0, J = 0, $ = 0, Le = "";
function gt(e, t, r, n, s, i, a, c) {
  return { value: e, root: t, parent: r, type: n, props: s, children: i, line: mt, column: De, length: a, return: "", siblings: c };
}
function pe(e, t) {
  return Ot(gt("", null, null, "", null, null, 0, e.siblings), e, { length: -e.length }, t);
}
function _e(e) {
  for (; e.root; )
    e = pe(e.root, { children: [e] });
  We(e, e.siblings);
}
function Zn() {
  return $;
}
function Qn() {
  return $ = J > 0 ? z(Le, --J) : 0, De--, $ === 10 && (De = 1, mt--), $;
}
function ee() {
  return $ = J < Nr ? z(Le, J++) : 0, De++, $ === 10 && (De = 1, mt++), $;
}
function Ce() {
  return z(Le, J);
}
function st() {
  return J;
}
function yt(e, t) {
  return je(Le, e, t);
}
function Tt(e) {
  switch (e) {
    // \0 \t \n \r \s whitespace token
    case 0:
    case 9:
    case 10:
    case 13:
    case 32:
      return 5;
    // ! + , / > @ ~ isolate token
    case 33:
    case 43:
    case 44:
    case 47:
    case 62:
    case 64:
    case 126:
    // ; { } breakpoint token
    case 59:
    case 123:
    case 125:
      return 4;
    // : accompanied token
    case 58:
      return 3;
    // " ' ( [ opening delimit token
    case 34:
    case 39:
    case 40:
    case 91:
      return 2;
    // ) ] closing delimit token
    case 41:
    case 93:
      return 1;
  }
  return 0;
}
function es(e) {
  return mt = De = 1, Nr = ne(Le = e), J = 0, [];
}
function ts(e) {
  return Le = "", e;
}
function xt(e) {
  return Or(yt(J - 1, Nt(e === 91 ? e + 2 : e === 40 ? e + 1 : e)));
}
function rs(e) {
  for (; ($ = Ce()) && $ < 33; )
    ee();
  return Tt(e) > 2 || Tt($) > 3 ? "" : " ";
}
function ns(e, t) {
  for (; --t && ee() && !($ < 48 || $ > 102 || $ > 57 && $ < 65 || $ > 70 && $ < 97); )
    ;
  return yt(e, st() + (t < 6 && Ce() == 32 && ee() == 32));
}
function Nt(e) {
  for (; ee(); )
    switch ($) {
      // ] ) " '
      case e:
        return J;
      // " '
      case 34:
      case 39:
        e !== 34 && e !== 39 && Nt($);
        break;
      // (
      case 40:
        e === 41 && Nt(e);
        break;
      // \
      case 92:
        ee();
        break;
    }
  return J;
}
function ss(e, t) {
  for (; ee() && e + $ !== 57; )
    if (e + $ === 84 && Ce() === 47)
      break;
  return "/*" + yt(t, J - 1) + "*" + $t(e === 47 ? e : ee());
}
function os(e) {
  for (; !Tt(Ce()); )
    ee();
  return yt(e, J);
}
function is(e) {
  return ts(ot("", null, null, null, [""], e = es(e), 0, [0], e));
}
function ot(e, t, r, n, s, i, a, c, l) {
  for (var d = 0, u = 0, b = a, y = 0, f = 0, w = 0, k = 1, C = 1, A = 1, E = 0, h = "", p = s, x = i, S = n, v = h; C; )
    switch (w = E, E = ee()) {
      // (
      case 40:
        if (w != 108 && z(v, b - 1) == 58) {
          nt(v += T(xt(E), "&", "&\f"), "&\f", Ar(d ? c[d - 1] : 0)) != -1 && (A = -1);
          break;
        }
      // " ' [
      case 34:
      case 39:
      case 91:
        v += xt(E);
        break;
      // \t \n \r \s
      case 9:
      case 10:
      case 13:
      case 32:
        v += rs(w);
        break;
      // \
      case 92:
        v += ns(st() - 1, 7);
        continue;
      // /
      case 47:
        switch (Ce()) {
          case 42:
          case 47:
            We(as(ss(ee(), st()), t, r, l), l);
            break;
          default:
            v += "/";
        }
        break;
      // {
      case 123 * k:
        c[d++] = ne(v) * A;
      // } ; \0
      case 125 * k:
      case 59:
      case 0:
        switch (E) {
          // \0 }
          case 0:
          case 125:
            C = 0;
          // ;
          case 59 + u:
            A == -1 && (v = T(v, /\f/g, "")), f > 0 && ne(v) - b && We(f > 32 ? tr(v + ";", n, r, b - 1, l) : tr(T(v, " ", "") + ";", n, r, b - 2, l), l);
            break;
          // @ ;
          case 59:
            v += ";";
          // { rule/at-rule
          default:
            if (We(S = er(v, t, r, d, u, s, c, h, p = [], x = [], b, i), i), E === 123)
              if (u === 0)
                ot(v, t, S, S, p, i, b, c, x);
              else
                switch (y === 99 && z(v, 3) === 110 ? 100 : y) {
                  // d l m s
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    ot(e, S, S, n && We(er(e, S, S, 0, 0, s, c, h, s, p = [], b, x), x), s, x, b, c, n ? p : x);
                    break;
                  default:
                    ot(v, S, S, S, [""], x, 0, c, x);
                }
        }
        d = u = f = 0, k = A = 1, h = v = "", b = a;
        break;
      // :
      case 58:
        b = 1 + ne(v), f = w;
      default:
        if (k < 1) {
          if (E == 123)
            --k;
          else if (E == 125 && k++ == 0 && Qn() == 125)
            continue;
        }
        switch (v += $t(E), E * k) {
          // &
          case 38:
            A = u > 0 ? 1 : (v += "\f", -1);
            break;
          // ,
          case 44:
            c[d++] = (ne(v) - 1) * A, A = 1;
            break;
          // @
          case 64:
            Ce() === 45 && (v += xt(ee())), y = Ce(), u = b = ne(h = v += os(st())), E++;
            break;
          // -
          case 45:
            w === 45 && ne(v) == 2 && (k = 0);
        }
    }
  return i;
}
function er(e, t, r, n, s, i, a, c, l, d, u, b) {
  for (var y = s - 1, f = s === 0 ? i : [""], w = Tr(f), k = 0, C = 0, A = 0; k < n; ++k)
    for (var E = 0, h = je(e, y + 1, y = Ar(C = a[k])), p = e; E < w; ++E)
      (p = Or(C > 0 ? f[E] + " " + h : T(h, /&\f/g, f[E]))) && (l[A++] = p);
  return gt(e, t, r, s === 0 ? ht : c, l, d, u, b);
}
function as(e, t, r, n) {
  return gt(e, t, r, kr, $t(Zn()), je(e, 2, -2), 0, n);
}
function tr(e, t, r, n, s) {
  return gt(e, t, r, It, je(e, 0, n), je(e, n + 1, -1), n, s);
}
function Fr(e, t, r) {
  switch (Xn(e, t)) {
    // color-adjust
    case 5103:
      return F + "print-" + e + e;
    // animation, animation-(delay|direction|duration|fill-mode|iteration-count|name|play-state|timing-function)
    case 5737:
    case 4201:
    case 3177:
    case 3433:
    case 1641:
    case 4457:
    case 2921:
    // text-decoration, filter, clip-path, backface-visibility, column, box-decoration-break
    case 5572:
    case 6356:
    case 5844:
    case 3191:
    case 6645:
    case 3005:
    // mask, mask-image, mask-(mode|clip|size), mask-(repeat|origin), mask-position, mask-composite,
    case 6391:
    case 5879:
    case 5623:
    case 6135:
    case 4599:
    case 4855:
    // background-clip, columns, column-(count|fill|gap|rule|rule-color|rule-style|rule-width|span|width)
    case 4215:
    case 6389:
    case 5109:
    case 5365:
    case 5621:
    case 3829:
      return F + e + e;
    // tab-size
    case 4789:
      return He + e + e;
    // appearance, user-select, transform, hyphens, text-size-adjust
    case 5349:
    case 4246:
    case 4810:
    case 6968:
    case 2756:
      return F + e + He + e + _ + e + e;
    // writing-mode
    case 5936:
      switch (z(e, t + 11)) {
        // vertical-l(r)
        case 114:
          return F + e + _ + T(e, /[svh]\w+-[tblr]{2}/, "tb") + e;
        // vertical-r(l)
        case 108:
          return F + e + _ + T(e, /[svh]\w+-[tblr]{2}/, "tb-rl") + e;
        // horizontal(-)tb
        case 45:
          return F + e + _ + T(e, /[svh]\w+-[tblr]{2}/, "lr") + e;
      }
    // flex, flex-direction, scroll-snap-type, writing-mode
    case 6828:
    case 4268:
    case 2903:
      return F + e + _ + e + e;
    // order
    case 6165:
      return F + e + _ + "flex-" + e + e;
    // align-items
    case 5187:
      return F + e + T(e, /(\w+).+(:[^]+)/, F + "box-$1$2" + _ + "flex-$1$2") + e;
    // align-self
    case 5443:
      return F + e + _ + "flex-item-" + T(e, /flex-|-self/g, "") + (ce(e, /flex-|baseline/) ? "" : _ + "grid-row-" + T(e, /flex-|-self/g, "")) + e;
    // align-content
    case 4675:
      return F + e + _ + "flex-line-pack" + T(e, /align-content|flex-|-self/g, "") + e;
    // flex-shrink
    case 5548:
      return F + e + _ + T(e, "shrink", "negative") + e;
    // flex-basis
    case 5292:
      return F + e + _ + T(e, "basis", "preferred-size") + e;
    // flex-grow
    case 6060:
      return F + "box-" + T(e, "-grow", "") + F + e + _ + T(e, "grow", "positive") + e;
    // transition
    case 4554:
      return F + T(e, /([^-])(transform)/g, "$1" + F + "$2") + e;
    // cursor
    case 6187:
      return T(T(T(e, /(zoom-|grab)/, F + "$1"), /(image-set)/, F + "$1"), e, "") + e;
    // background, background-image
    case 5495:
    case 3959:
      return T(e, /(image-set\([^]*)/, F + "$1$`$1");
    // justify-content
    case 4968:
      return T(T(e, /(.+:)(flex-)?(.*)/, F + "box-pack:$3" + _ + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + F + e + e;
    // justify-self
    case 4200:
      if (!ce(e, /flex-|baseline/)) return _ + "grid-column-align" + je(e, t) + e;
      break;
    // grid-template-(columns|rows)
    case 2592:
    case 3360:
      return _ + T(e, "template-", "") + e;
    // grid-(row|column)-start
    case 4384:
    case 3616:
      return r && r.some(function(n, s) {
        return t = s, ce(n.props, /grid-\w+-end/);
      }) ? ~nt(e + (r = r[t].value), "span", 0) ? e : _ + T(e, "-start", "") + e + _ + "grid-row-span:" + (~nt(r, "span", 0) ? ce(r, /\d+/) : +ce(r, /\d+/) - +ce(e, /\d+/)) + ";" : _ + T(e, "-start", "") + e;
    // grid-(row|column)-end
    case 4896:
    case 4128:
      return r && r.some(function(n) {
        return ce(n.props, /grid-\w+-start/);
      }) ? e : _ + T(T(e, "-end", "-span"), "span ", "") + e;
    // (margin|padding)-inline-(start|end)
    case 4095:
    case 3583:
    case 4068:
    case 2532:
      return T(e, /(.+)-inline(.+)/, F + "$1$2") + e;
    // (min|max)?(width|height|inline-size|block-size)
    case 8116:
    case 7059:
    case 5753:
    case 5535:
    case 5445:
    case 5701:
    case 4933:
    case 4677:
    case 5533:
    case 5789:
    case 5021:
    case 4765:
      if (ne(e) - 1 - t > 6)
        switch (z(e, t + 1)) {
          // (m)ax-content, (m)in-content
          case 109:
            if (z(e, t + 4) !== 45)
              break;
          // (f)ill-available, (f)it-content
          case 102:
            return T(e, /(.+:)(.+)-([^]+)/, "$1" + F + "$2-$3$1" + He + (z(e, t + 3) == 108 ? "$3" : "$2-$3")) + e;
          // (s)tretch
          case 115:
            return ~nt(e, "stretch", 0) ? Fr(T(e, "stretch", "fill-available"), t, r) + e : e;
        }
      break;
    // grid-(column|row)
    case 5152:
    case 5920:
      return T(e, /(.+?):(\d+)(\s*\/\s*(span)?\s*(\d+))?(.*)/, function(n, s, i, a, c, l, d) {
        return _ + s + ":" + i + d + (a ? _ + s + "-span:" + (c ? l : +l - +i) + d : "") + e;
      });
    // position: sticky
    case 4949:
      if (z(e, t + 6) === 121)
        return T(e, ":", ":" + F) + e;
      break;
    // display: (flex|inline-flex|grid|inline-grid)
    case 6444:
      switch (z(e, z(e, 14) === 45 ? 18 : 11)) {
        // (inline-)?fle(x)
        case 120:
          return T(e, /(.+:)([^;\s!]+)(;|(\s+)?!.+)?/, "$1" + F + (z(e, 14) === 45 ? "inline-" : "") + "box$3$1" + F + "$2$3$1" + _ + "$2box$3") + e;
        // (inline-)?gri(d)
        case 100:
          return T(e, ":", ":" + _) + e;
      }
      break;
    // scroll-margin, scroll-margin-(top|right|bottom|left)
    case 5719:
    case 2647:
    case 2135:
    case 3927:
    case 2391:
      return T(e, "scroll-", "scroll-snap-") + e;
  }
  return e;
}
function lt(e, t) {
  for (var r = "", n = 0; n < e.length; n++)
    r += t(e[n], n, e, t) || "";
  return r;
}
function cs(e, t, r, n) {
  switch (e.type) {
    case Kn:
      if (e.children.length) break;
    case Un:
    case It:
      return e.return = e.return || e.value;
    case kr:
      return "";
    case Cr:
      return e.return = e.value + "{" + lt(e.children, n) + "}";
    case ht:
      if (!ne(e.value = e.props.join(","))) return "";
  }
  return ne(r = lt(e.children, n)) ? e.return = e.value + "{" + r + "}" : "";
}
function ls(e) {
  var t = Tr(e);
  return function(r, n, s, i) {
    for (var a = "", c = 0; c < t; c++)
      a += e[c](r, n, s, i) || "";
    return a;
  };
}
function us(e) {
  return function(t) {
    t.root || (t = t.return) && e(t);
  };
}
function ds(e, t, r, n) {
  if (e.length > -1 && !e.return)
    switch (e.type) {
      case It:
        e.return = Fr(e.value, e.length, r);
        return;
      case Cr:
        return lt([pe(e, { value: T(e.value, "@", "@" + F) })], n);
      case ht:
        if (e.length)
          return Jn(r = e.props, function(s) {
            switch (ce(s, n = /(::plac\w+|:read-\w+)/)) {
              // :read-(only|write)
              case ":read-only":
              case ":read-write":
                _e(pe(e, { props: [T(s, /:(read-\w+)/, ":" + He + "$1")] })), _e(pe(e, { props: [s] })), Ot(e, { props: Qt(r, n) });
                break;
              // :placeholder
              case "::placeholder":
                _e(pe(e, { props: [T(s, /:(plac\w+)/, ":" + F + "input-$1")] })), _e(pe(e, { props: [T(s, /:(plac\w+)/, ":" + He + "$1")] })), _e(pe(e, { props: [T(s, /:(plac\w+)/, _ + "input-$1")] })), _e(pe(e, { props: [s] })), Ot(e, { props: Qt(r, n) });
                break;
            }
            return "";
          });
    }
}
var fs = {
  animationIterationCount: 1,
  aspectRatio: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
}, Oe = typeof process < "u" && process.env !== void 0 && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", Rr = "active", Vr = "data-styled-version", vt = "6.1.17", Mt = `/*!sc*/
`, ut = typeof window < "u" && "HTMLElement" in window, ps = !!(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env !== void 0 && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), rr = /invalid hook call/i, Qe = /* @__PURE__ */ new Set(), hs = function(e, t) {
  if (process.env.NODE_ENV !== "production") {
    var r = t ? ' with the id of "'.concat(t, '"') : "", n = "The component ".concat(e).concat(r, ` has been created dynamically.
`) + `You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.
See https://styled-components.com/docs/basics#define-styled-components-outside-of-the-render-method for more info.
`, s = console.error;
    try {
      var i = !0;
      console.error = function(a) {
        for (var c = [], l = 1; l < arguments.length; l++) c[l - 1] = arguments[l];
        rr.test(a) ? (i = !1, Qe.delete(n)) : s.apply(void 0, Pe([a], c, !1));
      }, br(), i && !Qe.has(n) && (console.warn(n), Qe.add(n));
    } catch (a) {
      rr.test(a.message) && Qe.delete(n);
    } finally {
      console.error = s;
    }
  }
}, bt = Object.freeze([]), Ie = Object.freeze({});
function ms(e, t, r) {
  return r === void 0 && (r = Ie), e.theme !== r.theme && e.theme || t || r.theme;
}
var Ft = /* @__PURE__ */ new Set(["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "tr", "track", "u", "ul", "use", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "tspan"]), gs = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, ys = /(^-|-$)/g;
function nr(e) {
  return e.replace(gs, "-").replace(ys, "");
}
var vs = /(a)(d)/gi, et = 52, sr = function(e) {
  return String.fromCharCode(e + (e > 25 ? 39 : 97));
};
function Rt(e) {
  var t, r = "";
  for (t = Math.abs(e); t > et; t = t / et | 0) r = sr(t % et) + r;
  return (sr(t % et) + r).replace(vs, "$1-$2");
}
var Et, _r = 5381, Se = function(e, t) {
  for (var r = t.length; r; ) e = 33 * e ^ t.charCodeAt(--r);
  return e;
}, Pr = function(e) {
  return Se(_r, e);
};
function bs(e) {
  return Rt(Pr(e) >>> 0);
}
function jr(e) {
  return process.env.NODE_ENV !== "production" && typeof e == "string" && e || e.displayName || e.name || "Component";
}
function kt(e) {
  return typeof e == "string" && (process.env.NODE_ENV === "production" || e.charAt(0) === e.charAt(0).toLowerCase());
}
var Dr = typeof Symbol == "function" && Symbol.for, Ir = Dr ? Symbol.for("react.memo") : 60115, ws = Dr ? Symbol.for("react.forward_ref") : 60112, Ss = { childContextTypes: !0, contextType: !0, contextTypes: !0, defaultProps: !0, displayName: !0, getDefaultProps: !0, getDerivedStateFromError: !0, getDerivedStateFromProps: !0, mixins: !0, propTypes: !0, type: !0 }, xs = { name: !0, length: !0, prototype: !0, caller: !0, callee: !0, arguments: !0, arity: !0 }, $r = { $$typeof: !0, compare: !0, defaultProps: !0, displayName: !0, propTypes: !0, type: !0 }, Es = ((Et = {})[ws] = { $$typeof: !0, render: !0, defaultProps: !0, displayName: !0, propTypes: !0 }, Et[Ir] = $r, Et);
function or(e) {
  return ("type" in (t = e) && t.type.$$typeof) === Ir ? $r : "$$typeof" in e ? Es[e.$$typeof] : Ss;
  var t;
}
var ks = Object.defineProperty, Cs = Object.getOwnPropertyNames, ir = Object.getOwnPropertySymbols, As = Object.getOwnPropertyDescriptor, Os = Object.getPrototypeOf, ar = Object.prototype;
function Mr(e, t, r) {
  if (typeof t != "string") {
    if (ar) {
      var n = Os(t);
      n && n !== ar && Mr(e, n, r);
    }
    var s = Cs(t);
    ir && (s = s.concat(ir(t)));
    for (var i = or(e), a = or(t), c = 0; c < s.length; ++c) {
      var l = s[c];
      if (!(l in xs || r && r[l] || a && l in a || i && l in i)) {
        var d = As(t, l);
        try {
          ks(e, l, d);
        } catch {
        }
      }
    }
  }
  return e;
}
function $e(e) {
  return typeof e == "function";
}
function Lt(e) {
  return typeof e == "object" && "styledComponentId" in e;
}
function ke(e, t) {
  return e && t ? "".concat(e, " ").concat(t) : e || t || "";
}
function cr(e, t) {
  if (e.length === 0) return "";
  for (var r = e[0], n = 1; n < e.length; n++) r += e[n];
  return r;
}
function Me(e) {
  return e !== null && typeof e == "object" && e.constructor.name === Object.name && !("props" in e && e.$$typeof);
}
function Vt(e, t, r) {
  if (r === void 0 && (r = !1), !r && !Me(e) && !Array.isArray(e)) return t;
  if (Array.isArray(t)) for (var n = 0; n < t.length; n++) e[n] = Vt(e[n], t[n]);
  else if (Me(t)) for (var n in t) e[n] = Vt(e[n], t[n]);
  return e;
}
function zt(e, t) {
  Object.defineProperty(e, "toString", { value: t });
}
var Ts = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
`, 18: "ThemeProvider: Please make sure your useTheme hook is within a `<ThemeProvider>`" } : {};
function Ns() {
  for (var e = [], t = 0; t < arguments.length; t++) e[t] = arguments[t];
  for (var r = e[0], n = [], s = 1, i = e.length; s < i; s += 1) n.push(e[s]);
  return n.forEach(function(a) {
    r = r.replace(/%[a-z]/, a);
  }), r;
}
function ze(e) {
  for (var t = [], r = 1; r < arguments.length; r++) t[r - 1] = arguments[r];
  return process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/errors.md#".concat(e, " for more information.").concat(t.length > 0 ? " Args: ".concat(t.join(", ")) : "")) : new Error(Ns.apply(void 0, Pe([Ts[e]], t, !1)).trim());
}
var Fs = function() {
  function e(t) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = t;
  }
  return e.prototype.indexOfGroup = function(t) {
    for (var r = 0, n = 0; n < t; n++) r += this.groupSizes[n];
    return r;
  }, e.prototype.insertRules = function(t, r) {
    if (t >= this.groupSizes.length) {
      for (var n = this.groupSizes, s = n.length, i = s; t >= i; ) if ((i <<= 1) < 0) throw ze(16, "".concat(t));
      this.groupSizes = new Uint32Array(i), this.groupSizes.set(n), this.length = i;
      for (var a = s; a < i; a++) this.groupSizes[a] = 0;
    }
    for (var c = this.indexOfGroup(t + 1), l = (a = 0, r.length); a < l; a++) this.tag.insertRule(c, r[a]) && (this.groupSizes[t]++, c++);
  }, e.prototype.clearGroup = function(t) {
    if (t < this.length) {
      var r = this.groupSizes[t], n = this.indexOfGroup(t), s = n + r;
      this.groupSizes[t] = 0;
      for (var i = n; i < s; i++) this.tag.deleteRule(n);
    }
  }, e.prototype.getGroup = function(t) {
    var r = "";
    if (t >= this.length || this.groupSizes[t] === 0) return r;
    for (var n = this.groupSizes[t], s = this.indexOfGroup(t), i = s + n, a = s; a < i; a++) r += "".concat(this.tag.getRule(a)).concat(Mt);
    return r;
  }, e;
}(), Rs = 1 << 30, it = /* @__PURE__ */ new Map(), dt = /* @__PURE__ */ new Map(), at = 1, tt = function(e) {
  if (it.has(e)) return it.get(e);
  for (; dt.has(at); ) at++;
  var t = at++;
  if (process.env.NODE_ENV !== "production" && ((0 | t) < 0 || t > Rs)) throw ze(16, "".concat(t));
  return it.set(e, t), dt.set(t, e), t;
}, Vs = function(e, t) {
  at = t + 1, it.set(e, t), dt.set(t, e);
}, _s = "style[".concat(Oe, "][").concat(Vr, '="').concat(vt, '"]'), Ps = new RegExp("^".concat(Oe, '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)')), js = function(e, t, r) {
  for (var n, s = r.split(","), i = 0, a = s.length; i < a; i++) (n = s[i]) && e.registerName(t, n);
}, Ds = function(e, t) {
  for (var r, n = ((r = t.textContent) !== null && r !== void 0 ? r : "").split(Mt), s = [], i = 0, a = n.length; i < a; i++) {
    var c = n[i].trim();
    if (c) {
      var l = c.match(Ps);
      if (l) {
        var d = 0 | parseInt(l[1], 10), u = l[2];
        d !== 0 && (Vs(u, d), js(e, u, l[3]), e.getTag().insertRules(d, s)), s.length = 0;
      } else s.push(c);
    }
  }
}, lr = function(e) {
  for (var t = document.querySelectorAll(_s), r = 0, n = t.length; r < n; r++) {
    var s = t[r];
    s && s.getAttribute(Oe) !== Rr && (Ds(e, s), s.parentNode && s.parentNode.removeChild(s));
  }
};
function Is() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}
var Lr = function(e) {
  var t = document.head, r = e || t, n = document.createElement("style"), s = function(c) {
    var l = Array.from(c.querySelectorAll("style[".concat(Oe, "]")));
    return l[l.length - 1];
  }(r), i = s !== void 0 ? s.nextSibling : null;
  n.setAttribute(Oe, Rr), n.setAttribute(Vr, vt);
  var a = Is();
  return a && n.setAttribute("nonce", a), r.insertBefore(n, i), n;
}, $s = function() {
  function e(t) {
    this.element = Lr(t), this.element.appendChild(document.createTextNode("")), this.sheet = function(r) {
      if (r.sheet) return r.sheet;
      for (var n = document.styleSheets, s = 0, i = n.length; s < i; s++) {
        var a = n[s];
        if (a.ownerNode === r) return a;
      }
      throw ze(17);
    }(this.element), this.length = 0;
  }
  return e.prototype.insertRule = function(t, r) {
    try {
      return this.sheet.insertRule(r, t), this.length++, !0;
    } catch {
      return !1;
    }
  }, e.prototype.deleteRule = function(t) {
    this.sheet.deleteRule(t), this.length--;
  }, e.prototype.getRule = function(t) {
    var r = this.sheet.cssRules[t];
    return r && r.cssText ? r.cssText : "";
  }, e;
}(), Ms = function() {
  function e(t) {
    this.element = Lr(t), this.nodes = this.element.childNodes, this.length = 0;
  }
  return e.prototype.insertRule = function(t, r) {
    if (t <= this.length && t >= 0) {
      var n = document.createTextNode(r);
      return this.element.insertBefore(n, this.nodes[t] || null), this.length++, !0;
    }
    return !1;
  }, e.prototype.deleteRule = function(t) {
    this.element.removeChild(this.nodes[t]), this.length--;
  }, e.prototype.getRule = function(t) {
    return t < this.length ? this.nodes[t].textContent : "";
  }, e;
}(), Ls = function() {
  function e(t) {
    this.rules = [], this.length = 0;
  }
  return e.prototype.insertRule = function(t, r) {
    return t <= this.length && (this.rules.splice(t, 0, r), this.length++, !0);
  }, e.prototype.deleteRule = function(t) {
    this.rules.splice(t, 1), this.length--;
  }, e.prototype.getRule = function(t) {
    return t < this.length ? this.rules[t] : "";
  }, e;
}(), ur = ut, zs = { isServer: !ut, useCSSOMInjection: !ps }, zr = function() {
  function e(t, r, n) {
    t === void 0 && (t = Ie), r === void 0 && (r = {});
    var s = this;
    this.options = G(G({}, zs), t), this.gs = r, this.names = new Map(n), this.server = !!t.isServer, !this.server && ut && ur && (ur = !1, lr(this)), zt(this, function() {
      return function(i) {
        for (var a = i.getTag(), c = a.length, l = "", d = function(b) {
          var y = function(A) {
            return dt.get(A);
          }(b);
          if (y === void 0) return "continue";
          var f = i.names.get(y), w = a.getGroup(b);
          if (f === void 0 || !f.size || w.length === 0) return "continue";
          var k = "".concat(Oe, ".g").concat(b, '[id="').concat(y, '"]'), C = "";
          f !== void 0 && f.forEach(function(A) {
            A.length > 0 && (C += "".concat(A, ","));
          }), l += "".concat(w).concat(k, '{content:"').concat(C, '"}').concat(Mt);
        }, u = 0; u < c; u++) d(u);
        return l;
      }(s);
    });
  }
  return e.registerId = function(t) {
    return tt(t);
  }, e.prototype.rehydrate = function() {
    !this.server && ut && lr(this);
  }, e.prototype.reconstructWithOptions = function(t, r) {
    return r === void 0 && (r = !0), new e(G(G({}, this.options), t), this.gs, r && this.names || void 0);
  }, e.prototype.allocateGSInstance = function(t) {
    return this.gs[t] = (this.gs[t] || 0) + 1;
  }, e.prototype.getTag = function() {
    return this.tag || (this.tag = (t = function(r) {
      var n = r.useCSSOMInjection, s = r.target;
      return r.isServer ? new Ls(s) : n ? new $s(s) : new Ms(s);
    }(this.options), new Fs(t)));
    var t;
  }, e.prototype.hasNameForId = function(t, r) {
    return this.names.has(t) && this.names.get(t).has(r);
  }, e.prototype.registerName = function(t, r) {
    if (tt(t), this.names.has(t)) this.names.get(t).add(r);
    else {
      var n = /* @__PURE__ */ new Set();
      n.add(r), this.names.set(t, n);
    }
  }, e.prototype.insertRules = function(t, r, n) {
    this.registerName(t, r), this.getTag().insertRules(tt(t), n);
  }, e.prototype.clearNames = function(t) {
    this.names.has(t) && this.names.get(t).clear();
  }, e.prototype.clearRules = function(t) {
    this.getTag().clearGroup(tt(t)), this.clearNames(t);
  }, e.prototype.clearTag = function() {
    this.tag = void 0;
  }, e;
}(), Bs = /&/g, qs = /^\s*\/\/.*$/gm;
function Br(e, t) {
  return e.map(function(r) {
    return r.type === "rule" && (r.value = "".concat(t, " ").concat(r.value), r.value = r.value.replaceAll(",", ",".concat(t, " ")), r.props = r.props.map(function(n) {
      return "".concat(t, " ").concat(n);
    })), Array.isArray(r.children) && r.type !== "@keyframes" && (r.children = Br(r.children, t)), r;
  });
}
function Ys(e) {
  var t, r, n, s = Ie, i = s.options, a = i === void 0 ? Ie : i, c = s.plugins, l = c === void 0 ? bt : c, d = function(y, f, w) {
    return w.startsWith(r) && w.endsWith(r) && w.replaceAll(r, "").length > 0 ? ".".concat(t) : y;
  }, u = l.slice();
  u.push(function(y) {
    y.type === ht && y.value.includes("&") && (y.props[0] = y.props[0].replace(Bs, r).replace(n, d));
  }), a.prefix && u.push(ds), u.push(cs);
  var b = function(y, f, w, k) {
    f === void 0 && (f = ""), w === void 0 && (w = ""), k === void 0 && (k = "&"), t = k, r = f, n = new RegExp("\\".concat(r, "\\b"), "g");
    var C = y.replace(qs, ""), A = is(w || f ? "".concat(w, " ").concat(f, " { ").concat(C, " }") : C);
    a.namespace && (A = Br(A, a.namespace));
    var E = [];
    return lt(A, ls(u.concat(us(function(h) {
      return E.push(h);
    })))), E;
  };
  return b.hash = l.length ? l.reduce(function(y, f) {
    return f.name || ze(15), Se(y, f.name);
  }, _r).toString() : "", b;
}
var Ws = new zr(), _t = Ys(), qr = H.createContext({ shouldForwardProp: void 0, styleSheet: Ws, stylis: _t });
qr.Consumer;
H.createContext(void 0);
function dr() {
  return ft(qr);
}
var fr = function() {
  function e(t, r) {
    var n = this;
    this.inject = function(s, i) {
      i === void 0 && (i = _t);
      var a = n.name + i.hash;
      s.hasNameForId(n.id, a) || s.insertRules(n.id, a, i(n.rules, a, "@keyframes"));
    }, this.name = t, this.id = "sc-keyframes-".concat(t), this.rules = r, zt(this, function() {
      throw ze(12, String(n.name));
    });
  }
  return e.prototype.getName = function(t) {
    return t === void 0 && (t = _t), this.name + t.hash;
  }, e;
}(), Hs = function(e) {
  return e >= "A" && e <= "Z";
};
function pr(e) {
  for (var t = "", r = 0; r < e.length; r++) {
    var n = e[r];
    if (r === 1 && n === "-" && e[0] === "-") return e;
    Hs(n) ? t += "-" + n.toLowerCase() : t += n;
  }
  return t.startsWith("ms-") ? "-" + t : t;
}
var Yr = function(e) {
  return e == null || e === !1 || e === "";
}, Wr = function(e) {
  var t, r, n = [];
  for (var s in e) {
    var i = e[s];
    e.hasOwnProperty(s) && !Yr(i) && (Array.isArray(i) && i.isCss || $e(i) ? n.push("".concat(pr(s), ":"), i, ";") : Me(i) ? n.push.apply(n, Pe(Pe(["".concat(s, " {")], Wr(i), !1), ["}"], !1)) : n.push("".concat(pr(s), ": ").concat((t = s, (r = i) == null || typeof r == "boolean" || r === "" ? "" : typeof r != "number" || r === 0 || t in fs || t.startsWith("--") ? String(r).trim() : "".concat(r, "px")), ";")));
  }
  return n;
};
function Ae(e, t, r, n) {
  if (Yr(e)) return [];
  if (Lt(e)) return [".".concat(e.styledComponentId)];
  if ($e(e)) {
    if (!$e(i = e) || i.prototype && i.prototype.isReactComponent || !t) return [e];
    var s = e(t);
    return process.env.NODE_ENV === "production" || typeof s != "object" || Array.isArray(s) || s instanceof fr || Me(s) || s === null || console.error("".concat(jr(e), " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.")), Ae(s, t, r, n);
  }
  var i;
  return e instanceof fr ? r ? (e.inject(r, n), [e.getName(n)]) : [e] : Me(e) ? Wr(e) : Array.isArray(e) ? Array.prototype.concat.apply(bt, e.map(function(a) {
    return Ae(a, t, r, n);
  })) : [e.toString()];
}
function Gs(e) {
  for (var t = 0; t < e.length; t += 1) {
    var r = e[t];
    if ($e(r) && !Lt(r)) return !1;
  }
  return !0;
}
var Us = Pr(vt), Ks = function() {
  function e(t, r, n) {
    this.rules = t, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (n === void 0 || n.isStatic) && Gs(t), this.componentId = r, this.baseHash = Se(Us, r), this.baseStyle = n, zr.registerId(r);
  }
  return e.prototype.generateAndInjectStyles = function(t, r, n) {
    var s = this.baseStyle ? this.baseStyle.generateAndInjectStyles(t, r, n) : "";
    if (this.isStatic && !n.hash) if (this.staticRulesId && r.hasNameForId(this.componentId, this.staticRulesId)) s = ke(s, this.staticRulesId);
    else {
      var i = cr(Ae(this.rules, t, r, n)), a = Rt(Se(this.baseHash, i) >>> 0);
      if (!r.hasNameForId(this.componentId, a)) {
        var c = n(i, ".".concat(a), void 0, this.componentId);
        r.insertRules(this.componentId, a, c);
      }
      s = ke(s, a), this.staticRulesId = a;
    }
    else {
      for (var l = Se(this.baseHash, n.hash), d = "", u = 0; u < this.rules.length; u++) {
        var b = this.rules[u];
        if (typeof b == "string") d += b, process.env.NODE_ENV !== "production" && (l = Se(l, b));
        else if (b) {
          var y = cr(Ae(b, t, r, n));
          l = Se(l, y + u), d += y;
        }
      }
      if (d) {
        var f = Rt(l >>> 0);
        r.hasNameForId(this.componentId, f) || r.insertRules(this.componentId, f, n(d, ".".concat(f), void 0, this.componentId)), s = ke(s, f);
      }
    }
    return s;
  }, e;
}(), Hr = H.createContext(void 0);
Hr.Consumer;
var Ct = {}, hr = /* @__PURE__ */ new Set();
function Xs(e, t, r) {
  var n = Lt(e), s = e, i = !kt(e), a = t.attrs, c = a === void 0 ? bt : a, l = t.componentId, d = l === void 0 ? function(p, x) {
    var S = typeof p != "string" ? "sc" : nr(p);
    Ct[S] = (Ct[S] || 0) + 1;
    var v = "".concat(S, "-").concat(bs(vt + S + Ct[S]));
    return x ? "".concat(x, "-").concat(v) : v;
  }(t.displayName, t.parentComponentId) : l, u = t.displayName, b = u === void 0 ? function(p) {
    return kt(p) ? "styled.".concat(p) : "Styled(".concat(jr(p), ")");
  }(e) : u, y = t.displayName && t.componentId ? "".concat(nr(t.displayName), "-").concat(t.componentId) : t.componentId || d, f = n && s.attrs ? s.attrs.concat(c).filter(Boolean) : c, w = t.shouldForwardProp;
  if (n && s.shouldForwardProp) {
    var k = s.shouldForwardProp;
    if (t.shouldForwardProp) {
      var C = t.shouldForwardProp;
      w = function(p, x) {
        return k(p, x) && C(p, x);
      };
    } else w = k;
  }
  var A = new Ks(r, y, n ? s.componentStyle : void 0);
  function E(p, x) {
    return function(S, v, M) {
      var I = S.attrs, j = S.componentStyle, R = S.defaultProps, V = S.foldedComponentIds, L = S.styledComponentId, U = S.target, Z = H.useContext(Hr), Te = dr(), he = S.shouldForwardProp || Te.shouldForwardProp;
      process.env.NODE_ENV !== "production" && Gt(L);
      var Ne = ms(v, Z, R) || Ie, B = function(ge, de, ye) {
        for (var oe, re = G(G({}, de), { className: void 0, theme: ye }), Fe = 0; Fe < ge.length; Fe += 1) {
          var ve = $e(oe = ge[Fe]) ? oe(re) : oe;
          for (var X in ve) re[X] = X === "className" ? ke(re[X], ve[X]) : X === "style" ? G(G({}, re[X]), ve[X]) : ve[X];
        }
        return de.className && (re.className = ke(re.className, de.className)), re;
      }(I, v, Ne), ue = B.as || U, K = {};
      for (var W in B) B[W] === void 0 || W[0] === "$" || W === "as" || W === "theme" && B.theme === Ne || (W === "forwardedAs" ? K.as = B.forwardedAs : he && !he(W, ue) || (K[W] = B[W], he || process.env.NODE_ENV !== "development" || Gn(W) || hr.has(W) || !Ft.has(ue) || (hr.add(W), console.warn('styled-components: it looks like an unknown prop "'.concat(W, '" is being sent through to the DOM, which will likely trigger a React console error. If you would like automatic filtering of unknown props, you can opt-into that behavior via `<StyleSheetManager shouldForwardProp={...}>` (connect an API like `@emotion/is-prop-valid`) or consider using transient props (`$` prefix for automatic filtering.)')))));
      var me = function(ge, de) {
        var ye = dr(), oe = ge.generateAndInjectStyles(de, ye.styleSheet, ye.stylis);
        return process.env.NODE_ENV !== "production" && Gt(oe), oe;
      }(j, B);
      process.env.NODE_ENV !== "production" && S.warnTooManyClasses && S.warnTooManyClasses(me);
      var se = ke(V, L);
      return me && (se += " " + me), B.className && (se += " " + B.className), K[kt(ue) && !Ft.has(ue) ? "class" : "className"] = se, M && (K.ref = M), en(ue, K);
    }(h, p, x);
  }
  E.displayName = b;
  var h = H.forwardRef(E);
  return h.attrs = f, h.componentStyle = A, h.displayName = b, h.shouldForwardProp = w, h.foldedComponentIds = n ? ke(s.foldedComponentIds, s.styledComponentId) : "", h.styledComponentId = y, h.target = n ? s.target : e, Object.defineProperty(h, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(p) {
    this._foldedDefaultProps = n ? function(x) {
      for (var S = [], v = 1; v < arguments.length; v++) S[v - 1] = arguments[v];
      for (var M = 0, I = S; M < I.length; M++) Vt(x, I[M], !0);
      return x;
    }({}, s.defaultProps, p) : p;
  } }), process.env.NODE_ENV !== "production" && (hs(b, y), h.warnTooManyClasses = /* @__PURE__ */ function(p, x) {
    var S = {}, v = !1;
    return function(M) {
      if (!v && (S[M] = !0, Object.keys(S).length >= 200)) {
        var I = x ? ' with the id of "'.concat(x, '"') : "";
        console.warn("Over ".concat(200, " classes were generated for component ").concat(p).concat(I, `.
`) + `Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), v = !0, S = {};
      }
    };
  }(b, y)), zt(h, function() {
    return ".".concat(h.styledComponentId);
  }), i && Mr(h, e, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0 }), h;
}
function mr(e, t) {
  for (var r = [e[0]], n = 0, s = t.length; n < s; n += 1) r.push(t[n], e[n + 1]);
  return r;
}
var gr = function(e) {
  return Object.assign(e, { isCss: !0 });
};
function Js(e) {
  for (var t = [], r = 1; r < arguments.length; r++) t[r - 1] = arguments[r];
  if ($e(e) || Me(e)) return gr(Ae(mr(bt, Pe([e], t, !0))));
  var n = e;
  return t.length === 0 && n.length === 1 && typeof n[0] == "string" ? Ae(n) : gr(Ae(mr(n, t)));
}
function Pt(e, t, r) {
  if (r === void 0 && (r = Ie), !t) throw ze(1, t);
  var n = function(s) {
    for (var i = [], a = 1; a < arguments.length; a++) i[a - 1] = arguments[a];
    return e(t, r, Js.apply(void 0, Pe([s], i, !1)));
  };
  return n.attrs = function(s) {
    return Pt(e, t, G(G({}, r), { attrs: Array.prototype.concat(r.attrs, s).filter(Boolean) }));
  }, n.withConfig = function(s) {
    return Pt(e, t, G(G({}, r), s));
  }, n;
}
var Gr = function(e) {
  return Pt(Xs, e);
}, Xe = Gr;
Ft.forEach(function(e) {
  Xe[e] = Gr(e);
});
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`);
var rt = "__sc-".concat(Oe, "__");
process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window[rt] || (window[rt] = 0), window[rt] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window[rt] += 1);
const Zs = Xe.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
`, yr = Xe.span`
  color: #f00;
  font-size: small;
  display: inline-block;
  margin: 0;
`, Qs = Xe.span`
  font-size: small;
  color: #828282;
  display: inline-block;
  margin: 0;
`, eo = tn(
  ({
    id: e,
    label: t,
    required: r,
    error: n,
    className: s = "",
    children: i,
    styles: a = {},
    labelClassName: c = "",
    labelStyles: l = {},
    type: d,
    description: u
  }) => {
    {
      const b = d !== "radio" && d !== "checkbox";
      return /* @__PURE__ */ P.jsxs(Zs, { className: s, style: a, children: [
        t && /* @__PURE__ */ P.jsxs(
          "label",
          {
            ...b ? { htmlFor: e } : {},
            className: c,
            style: l,
            children: [
              /* @__PURE__ */ P.jsx("span", { children: t }),
              " ",
              r && /* @__PURE__ */ P.jsx(yr, { children: "*" })
            ]
          }
        ),
        i,
        u && /* @__PURE__ */ P.jsx(Qs, { children: u }),
        n && /* @__PURE__ */ P.jsx(yr, { children: n })
      ] });
    }
  }
), we = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.InputField }))
), to = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.DatePickerField }))
), ro = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.GroupField }))
), no = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.SelectField }))
), so = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.RadioField }))
), oo = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.CheckboxField }))
), io = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.GridViewField }))
), ao = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.ContainerField }))
), co = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.TextareaField }))
), lo = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.SpacerField }))
), uo = te(
  () => import("./index-aa7hYACF.js").then((e) => ({ default: e.InputFileField }))
), Ur = {
  group: ro,
  text: we,
  number: we,
  email: we,
  tel: we,
  url: we,
  password: we,
  search: we,
  date: to,
  select: no,
  radio: so,
  checkbox: oo,
  gridview: io,
  container: ao,
  textarea: co,
  spacer: lo,
  file: uo
  // Add other fields here as needed
}, yo = (e, t) => {
  Ur[e] = t;
}, fo = () => /* @__PURE__ */ P.jsxs("div", { children: [
  /* @__PURE__ */ P.jsx(
    Zt,
    {
      height: 15,
      width: "25%",
      style: { opacity: 0.2, marginTop: "10px" }
    }
  ),
  /* @__PURE__ */ P.jsx(
    Zt,
    {
      height: 40,
      width: "100%",
      style: { opacity: 0.5, marginBottom: "15px" }
    }
  )
] }), po = ({
  showSkeletonLoading: e = !0,
  skeleton: t,
  ...r
}) => {
  const {
    processedProps: n,
    errors: s,
    isVisible: i
    // observer,
    // fieldId,
    // values,
    // validateField,
  } = Dn(r), [a, c] = le(), [l, d] = le(!0);
  Ge(() => {
    c(s);
  }, [s]), Ge(() => {
    d(i);
  }, [i]);
  const u = Ur[r.type];
  return !u || !l ? null : /* @__PURE__ */ P.jsx(
    rn,
    {
      fallback: e && t !== void 0 ? t : e ? /* @__PURE__ */ P.jsx(fo, {}) : null,
      children: /* @__PURE__ */ P.jsx(
        eo,
        {
          id: n.fieldId,
          label: n.label,
          required: n.required,
          error: a?.[n.fieldId],
          className: n.containerClassName,
          styles: n.containerStyles,
          labelClassName: n.labelClassName,
          labelStyles: n.labelStyles,
          type: n.type,
          description: n.description,
          children: /* @__PURE__ */ P.jsx(u, { ...n })
        }
      )
    }
  );
}, ho = Xe.button`
  background-color: #0457aa;
  border: none;
  color: white;
  padding: 8px 16px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  border-radius: 8px;
  cursor: pointer;
  margin: 12px 0;
`, vr = ({
  formData: e,
  onSubmit: t,
  isLoading: r,
  submitDetails: n,
  showSkeletonLoading: s = !0,
  skeleton: i
}) => {
  const { values: a, validateForm: c } = Ke(), l = (d) => {
    d.preventDefault();
    const u = c(e);
    t && t(a, u);
  };
  return /* @__PURE__ */ P.jsxs("form", { onSubmit: l, role: "form", children: [
    r && /* @__PURE__ */ P.jsx("div", {}),
    e.fields.map((d) => /* @__PURE__ */ P.jsx(
      po,
      {
        ...d,
        skeleton: i,
        showSkeletonLoading: s
      },
      d.fieldId
    )),
    n?.visibility !== !1 && /* @__PURE__ */ P.jsx(
      "div",
      {
        className: n?.containerClassName,
        style: n?.containerStyles,
        children: /* @__PURE__ */ P.jsx(ho, { type: "submit", disabled: r, children: r ? "Submitting..." : n?.text || "Submit" })
      }
    )
  ] });
}, vo = ({
  formId: e,
  formData: t,
  onSubmit: r,
  isLoading: n = !1,
  submitDetails: s,
  customProvider: i,
  showSkeletonLoading: a = !0,
  skeleton: c
}) => {
  const l = ft(ct), d = i || Bn;
  return l ? /* @__PURE__ */ P.jsx(
    vr,
    {
      formId: e,
      formData: t,
      onSubmit: r,
      isLoading: n,
      submitDetails: s,
      showSkeletonLoading: a,
      skeleton: c
    }
  ) : /* @__PURE__ */ P.jsx(d, { formSchema: t, formId: e, children: /* @__PURE__ */ P.jsx(
    vr,
    {
      formId: e,
      formData: t,
      onSubmit: r,
      isLoading: n,
      submitDetails: s,
      showSkeletonLoading: a,
      skeleton: c
    }
  ) });
};
export {
  po as D,
  eo as F,
  Ke as a,
  vo as b,
  Bn as c,
  Xe as d,
  ct as e,
  _n as f,
  Pn as g,
  In as h,
  jn as i,
  P as j,
  yo as r,
  Dn as u
};
